src/libutil/util.cc

```diff
@@ -1831,4 +1831,25 @@ void commonChildInit(Pipe & logPipe)
     close(fdDevNull);
 }
 
+/* Given two strings, return a pointer to the start of the first
+ * non-equivalent part of the first string. This is a non-owning reference to
+ * data owned by a, and must not outlive a. */
+const std::string_view removeStartingOverlap(const std::string & a_ref, const std::string & b_ref)
```

Comment:
Maybe we could make use of std::mismatch?

Reply:
Yep, works great - didn't even know that existed! 6ab209ac4